utilities/blob_db/blob_db_impl.cc

```diff
@@ -1655,8 +1658,8 @@ std::pair<bool, int64_t> BlobDBImpl::WaStats(bool aborted) {
 // DELETED in the LSM
 ////////////////////////////////////////////////////////////////////////////////
 Status BlobDBImpl::GCFileAndUpdateLSM(const std::shared_ptr<BlobFile>& bfptr,
```

Comment:
1. This function has grown too big. Lets work on splitting it up into smaller more manageable pieces, may be in a separate PR.
2. I also feel there are too many breaks and continues in the read-record loop, hindering the normal flow of understanding. It would be great if we can think of cleaning up that loop some more later.